### PR TITLE
Bump dependency versions to RTM

### PR DESCRIPTION
**Pull request by nguerrera**
Also bumped FluentAssertions versions to latest.

I am still working to get rid of the remaining manual copying, but separating the package upgrades from that here.

@eerhardt @dsplaisted @livarcocc

### CODE REVIEW

**Review by eerhardt — APPROVED** [review]

File: src/Dependencies/xUnit.net/xUnit.net.csproj [hunk 1]
```diff
@@ -44,13 +44,17 @@
       <CopyLocalAssembly Include="xunit.runner.utility.dotnet">
         <RelativePath>xunit.runner.utility\$(XUnitVersion)\lib\dotnet\xunit.runner.utility.dotnet.dll</RelativePath>
       </CopyLocalAssembly>
+      <CopyLocalAssembly Include="fluentassertions">
```
Comment (livarcocc) [inline src/Dependencies/xUnit.net/xUnit.net.csproj]: We need this because we are not using dotnet test yet, correct?

**Review by livarcocc — APPROVED** [review]

File: src/Dependencies/xUnit.net/xUnit.net.csproj (hunk 1 above)
Comment (nguerrera) [inline src/Dependencies/xUnit.net/xUnit.net.csproj]: Yes, I will fix it soon, but I'm splitting in to steps as it's easier to work through the issues.

File: build/Nuget/Microsoft.NET.Sdk.Nuget.targets [hunk 2]
```diff
@@ -8,7 +8,7 @@
       <PackageFile Include="$(OutDir)build\**\*.*" Folder="build" />
       <PackageFile Include="$(OutDir)buildCrossTargeting\**\*.*" Folder="buildCrossTargeting" />
       <PackageFile Include="$(OutDir)sdk\**\*.*" Folder="Sdk" />
-      <PackageFile Include="$(NuGet_Packages)\newtonsoft.json\6.0.4\lib\net45\Newtonsoft.Json.dll" Folder="tools\net46\Newtonsoft.Json.6.0.4" />
+      <PackageFile Include="$(NuGet_Packages)\newtonsoft.json\$(NewtonsoftJsonOnNetFrameworkVersion)\lib\net45\Newtonsoft.Json.dll" Folder="tools\net46\Newtonsoft.Json.6.0.4" Condition="'$(OS)' == 'Windows_NT'" />
```
Comment (eerhardt) [inline build/Nuget/Microsoft.NET.Sdk.Nuget.targets]: Would it make sense to use the `$(NewtonsoftJsonOnNetFrameworkVersion)` propery in the `Folder` path as well?  Feels weird to use the variable here, but still have the hard-coded version in the folder path.

Comment (nguerrera) [inline build/Nuget/Microsoft.NET.Sdk.Nuget.targets]: Yup. Will fix.

**Pull request merged**